README.md

```diff
@@ -28,31 +41,117 @@ make
 make install
 ```
 
-## Usage
-1) Create a YAML configuration file:
+## Usage Examples
+
+1) Create a complex YAML configuration:
+
+```yaml
+simulation:
+  parameters:
+    timestep: 0.01
+    max_iterations: 1000
+    tolerances:
+      - 1.0e-6
+      - 1.0e-8
+  output:
+    format: netcdf
+    variables: [temperature, pressure, velocity]
+    frequency: 100
+```
+
+2) Parse and access data:
+
+```fortran
+program simulation_setup
+    use fyaml
+
+    type(fyaml_doc) :: config
+    type(yaml_value) :: val
+    real(dp) :: timestep
+    character(len=:), allocatable, dimension(:) :: variables
+
+    ! Load configuration
+    call config%load("simulation.yaml")
+
+    ! Get scalar values using dot notation
+    timestep = config%get("simulation.parameters.timestep")%get_real()
```

Comment:
This is cool, but we could also think about adding a similar API to toml-f, with a `get_value(node, value)` subroutine that gets the correct type based on the type of `value` (with an interface block).

Reply:
Having some issues with creating an interface to do this. I suggest that we come back to this.